source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,117 @@
+.. _glossary:
+
+=========
+Glossary
+=========
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Attribute
+        A particular property or quality of a real object or an event.
+        Has a defined type and domain.
+
+    Classification
+        A supervised machine-learning problem to assign categorical :term:`labels<Label>`
+        to :term:`feature vectors<Feature vector>` describing input objects.
+
+    Dataset
+        A collection of :term:`instances<Instance>`.
+
+    Feature
+        A pair of :term:`attribute` information and a corresponding value.
+
+    Feature vector
+        A vector that encodes information about single real object or an event.
+        Contains at least one :term:`feature`.
+
+    Inference
+        A process of obtaining outputs from the :term:`model` created on the :term:`training` stage
+
+    Instance
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+    Label
+        :term:`Feature` of some object or event which is present on the :term:`training` stage but
+        unavaliable on the :term:`inference` stage.
+
+    Model
+        A result of :term:`training` machine-learning algorithm that stores information necessary to run
+        :term:`inference` on a new :term:`dataset`.
+
+    Observation
+        A synonym for :term:`instance`.
+
+    Prediction
+        - A synonym for :term:`inference`.
+        - An output of the :term:`trained<Training>` :term:`model` when provided with an input :term:`instance`.
+
+    Response
+        A value of a single :term:`label`. For example, response may be a flag
+        indicating that an observation belongs to a particular class.
+
+    Sample
+        A synonym for :term:`instance`.
+
+    Training
```

Comment:
Changed to "A search process of the best model" - is it more accurate?

Reply:
It's grammatically incorrect (search **for**)
But I also don't like this definition because it doesn't say anything about how you determine which model is the best. What exactly are you searching for? What do you have at the beginning of that search (e.g. you need a training set and a target variable).
I'd mention parametric and non-parametric methods here (q: should there be entries for them in the glossary?)

Reply:
Reformulated, please look. It is very hard to bring any details about this term because it is very common. I cannot even say that it is an optimization process, that's why I used the term "search process".